develop/devguide/Automation/Howto/Implementing_OnRequestScriptInfo_Entry_Point.md

```diff
@@ -0,0 +1,105 @@
+
+- When an omitted script parameter is used in the entry point logic, retrieving the script parameter is possible, but its value will be an empty string.
+- When an omitted dummy is used in the entry point logic, retrieving the dummy is possible, but it will refer to DMA ID -1 and element ID -1. Any actions that use the dummy will fail with an exception.
+- When an omitted memory file is used in the entry point logic, retrieving the memory file is possible, but it will refer to a linked file that is empty. Retrieving a value using the memory file will fail with an exception.
+
+## Starting the entry point
+
+To start the `OnRequestScriptInfo` entry point the below methods can be used. Within an Automation script a [subscript](#subscript) should be started. Using the [ExecuteScriptMessage](#executescriptmessage) will only allow 10 simultaneous executions at most.
+
+### Subscript
+
+To execute the `OnRequestScriptInfo` entry point within Automation, you have to use the [PrepareSubScript](xref:Skyline.DataMiner.Automation.Engine.OnRequestScriptInfo) method.
+
+In the following snippet, the entry point of script 'Script with entry point' will be started. As input data the 'Action' key with value 'RequestValues'. After executing the script's entry point synchronously (default behavior), the returned output is checked for the value of key 'ActionResult'.
+
+```csharp
+var input = new RequestScriptInfoInput
+{
+    Data = new Dictionary<string, string>
+    {
+        { "Action", "RequestValues" },
+    },
+};
+
+var subscriptOptions = engine.PrepareSubScript("Script with entry point", input);
+subscriptOptions.StartScript();
+
+if (subscriptOptions.Output?.Data is null ||
+    !subscriptOptions.Output.Data.TryGetValue("ActionResult", out string resultKeyValue))
+{
+    engine.ExitFail("Expected an ActionResult in the output of the subscript.");
+}
+```
+
+The script should be started synchronously. It will return a subscript options object with an `Output` property containing the information returned by the script. The `Input` property can be used to check or update the data sent to the script.
+
+### ExecuteScriptMessage
+
+The `ExecuteScriptMessage` can be used to trigger the entry point using an SLNet connection.
+
+In the following snippet, the entry point of script 'Script with entry point' will be started. As input data the 'Action' key with value 'RequestValues'. After executing the script's entry point synchronously, the returned output is checked for the value of key 'ActionResult'.
+
+```csharp
+var input = new RequestScriptInfoInput
+{
+    Data = new Dictionary<string, string>
+    {
+        { "Action", "RequestValues" },
+    },
+};
+
+var msg = new ExecuteScriptMessage
+{
+    ScriptName = "Script with entry point",
+    Options = new SA(new[] { "DEFER:FALSE" }),
```

Comment:
So we need to provide these options? Is it not sync by default?

Reply:
As discussed, async is the default I'm afraid.